### ISSUE TITLE
biblatex with style=philosophy-modern error

### ISSUE THREAD

**Issue by llde** [tectonic-typesetting/tectonic#39]
I was trying some latex source I had with tectonic, and I'm having an issue with biblatex with style=philosophy-modern option:

! Undefined control sequence.
l.1637 \ifblx@load@version@legacy

Someone on stackoverflow say it is caused by an old version of biblatex (or biblatex philosophy).

This is the project I tryied to compile: https://github.com/LucaMoschella/SEProject

**Comment by jtojnar** [comment]
Indeed, Tectonic uses old version, I also get an error, though a [different one](https://github.com/tectonic-typesetting/tectonic/issues/33#issuecomment-305229316).

```
$ sed -n 14,15p /usr/share/texmf-dist/tex/latex/biblatex/biblatex.sty
\def\abx@date{2016/12/05}
\def\abx@version{3.7}
$ sed -n 14,15p ~/.cache/Tectonic/files/d2/2a9bf2fd00b513129c0366c63bcd8ceadfa5b81144d394586c7448059f44a2
\def\abx@date{2016/05/14}
\def\abx@version{3.4}
```

**Comment by llde** [comment]
Proposal @pkgw : Allow to query CTAN (thought XML or JSON API) as a repo to detect when  certain libraries are outdated or missing in the tectonic bundle.

**Comment by pkgw** [comment]
First of all: as per issue #35, Tectonic is not currently compatible with `biblatex` using the `biber` backend, and it probably won't be for a long time. Sorry. But, this problem is a separate issue from that.

I can confirm that the current default bundle (`tlextras-2016.0r4`) distributes the `2016/05/14` version of this file. As far as I can verify for myself, I can also confirm that this version is the one that's distributed with TeXLive 2016, which is the intended behavior.

So:

1. There might be a problem in the bundle if it includes a "too-new" version of `philosophy-modern`
2. There might be a problem in the underyling TeXLive 2016 distribution that just has to be worked around
3. There might be a problem in this particular document where it tries to use a combination of packages that doesn't correspond to a locked version of TeXLive 2016.

Case (1) is the only one that I would consider a Tectonic bug — I am very strongly of the opinion that the underlying bundle should be a pristine-as-possible snapshot of a TeXLive distribution.

**Comment by jtojnar** [comment]
I undestand the necessity of using a stable TeXLive snapshot for reproducible builds, however, since TexLive creates a snapshot once a year, some packages are quite outdated already. It would be nice to be able to force the use of more recent package version in `Tectonic.toml`.

Hopefully, a new TexLive version will be [released soon](https://en.wikipedia.org/wiki/TeX_Live#Release_history) and these problems will be fixed then but for most of the year version override will be still necessary.

**Comment by pkgw** [comment]
I guess my view of the TeX world has been colored by my background as a working scientist — all of the TeX packages we use are generally old and super stable, so yearly updates seem just fine :-)

Some points:

- In cases like this, I've been envisioning that users should make a copy of whatever updated files they need and store them with their document. This recommendation needs to be documented somewhere, though, and possibly relayed to the user by the command-line program when appropriate.
- Perhaps there is a case for a "bleeding-edge" bundle that is updated more regularly from something like TeXLive master. One could envision writing an I/O backend that went right to the SVN server and used the `ls-R` files to find what it needed.

Anyway, these ideas are probably best discussed in separate GitHub issues. @llde, please close this one if you feel that it covers your specific problem.

**Comment by llde** [comment]
@pkgw  An issue with this way to go is that if some libraries are not included in the bundle they are currently not installable, and we are forced to use older library versions.
In practice this may be an issue if I want to use some functionality in a new library version not included in the bundle. 
I think that in this case aviability may be a better use case then reproducibility (even if not as default).

This doens't seem to be the biblatex with biber issue, becouse it's failing earlier. Also on TexLive 2016 is working correctly with Xelatex/PdfLAtex/Latex backends.  So maybe keep this bug open to further discuss what went wrong may be better then put everyting into the #35 cauldron.

**Comment by llde** [comment]
So much time passed that I didn't even remember this bug report existed. 
If I found other bugs, I will certainly report them.
At next time.
